src/plugins/kernel_install/utils.sh

```diff
@@ -550,32 +550,85 @@ function do_uninstall()
   fi
 }
 
+# Element existence check
+# Checks if element belongs to given array
+#
+# @target_element element being checked
+# @array target array
+# returns 1 if element is in array and 0 otherwise
+function is_in_array()
+{
+  local target_element="$1"
+  local array=("${!2}")
+
+  for element in "${array[@]}"; do
+    if [[ "$element" == "$target_element" ]]; then
+      return 1
+    fi
+  done
+  return 0
+}
+
+# Return available kernels
+#
+# @prefix Set a base prefix for searching for kernels.
+# @all_kernels List all available kernels, not only the ones installed by kw
+# returns array containing available kernels
+function process_installed_kernels()
+{
+  local all_kernels="$1"
+  local prefix="$2"
+  local kernels
+
```

Comment:
Just like in the function `is_in_array` prefer the use o functions that manipulate array references using `local -n` over functions that "globally" do this. If I am not mistaken, notice that you are creating an array `deduped_kernels` that "bypasses" the function API, in the sense that the caller must know that the function processed will be in this specific array.

My request is to define another argument `local -n _processed_installed_kernels="$3"` that will be declared by the caller, and it will passed it as reference to this function to store the processed installed kernels.